api/stse_device_authentication.c

```diff
@@ -1,182 +1,225 @@
+	PLAT_UI8 certificate_size_ui8[2];
+
+	/* - Check stsafe handler initialization */
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+
+
+	if (pCertificate_size == NULL)
+	{
+		return ( STSE_API_INVALID_PARAMETER );
+	}
+
+	ret = stse_data_storage_read_data_zone(
+			pSTSE,								/* STSAFE handler */
+			certificate_zone,					/* Certificate zone */
+			STSE_CERTIFICATE_SIZE_OFFSET_BYTES,	/* 2 bytes offset */
+			certificate_size_ui8,				/* Returned certificate size */
+			STSE_CERTIFICATE_SIZE_LENGTH,		/* Certificate size length 2 bytes */
+			0,									/* No maximum chunck size (No chunck at all) */
+			STSAFEA_NO_PROT);					/* No protection */
+
+	if(ret != STSE_OK)
+	{
+		return( ret );
+	}
+
+	*pCertificate_size = (((uint16_t)certificate_size_ui8[0]) << 8) + ((uint16_t)certificate_size_ui8[1]) + 4U;
+
+	return STSE_OK;
+}
+
+stse_ReturnCode_t stse_get_device_certificate(stse_Handler_t * pSTSE, PLAT_UI8 certificate_zone, PLAT_UI16 certificate_size, PLAT_UI8* pCertificate)
+{
+	volatile stse_ReturnCode_t ret = STSE_API_INVALID_PARAMETER;
+
+	/* - Check stsafe handler initialization */
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+
+	if (pCertificate == NULL)
+	{
+		return ( STSE_API_INVALID_PARAMETER );
+	}
+
+	ret = stse_data_storage_read_data_zone(
+			pSTSE,								/* STSAFE handler */
+			certificate_zone,					/* Certificate zone */
+			STSE_CERTIFICATE_OFFSET_BYTES,		/* 0 bytes offset */
+			pCertificate,						/* Returned certificate size */
+			certificate_size,					/* Certificate size length */
+			255,								/* No maximum chunck size (No chunck at all) */
+			STSAFEA_NO_PROT);					/* No protection */
+
+	if(ret != STSE_OK)
+	{
+		return( ret );
+	}
+
+	return STSE_OK;
+}
+
+stse_ReturnCode_t stse_device_authenticate(
+		stse_Handler_t * pSTSE,
+		const PLAT_UI8 *pRoot_CA_certificate,
+		PLAT_UI8 certificate_zone,
+		PLAT_UI8 priv_key_slot_number)
+{
+	stse_ReturnCode_t ret;
+	stse_certificate_t leaf_certificate;
+	PLAT_UI16 certificate_size;
+	PLAT_UI8 certificate_size_ui8[2];
+
+	if (pSTSE == NULL)
+	{
+		return( STSE_API_HANDLER_NOT_INITIALISED );
+	}
+
+	if (pRoot_CA_certificate == NULL)
+	{
+		return ( STSE_API_INVALID_PARAMETER );
+	}
+
+	ret = stse_data_storage_read_data_zone(
+			pSTSE,
+			certificate_zone,
+			STSE_CERTIFICATE_SIZE_OFFSET_BYTES,	/* 2 bytes offset */
+			certificate_size_ui8,				/* Returned certificate size */
+			STSE_CERTIFICATE_SIZE_LENGTH,		/* Certificate size length 2 bytes */
+			0,									/* No maximum chunck size (No chunck at all) */
+			STSAFEA_NO_PROT);					/* No protection */
+
+	if(ret != STSE_OK)
+	{
+		return ret;
+	}
+
+	certificate_size = (((uint16_t)certificate_size_ui8[0]) << 8) + ((uint16_t)certificate_size_ui8[1]) + 4U;
+
+	PLAT_UI8 certificate[certificate_size];
+
+	ret = stse_get_device_certificate(pSTSE, certificate_zone, certificate_size, certificate);
+	if (ret != STSE_OK)
+	{
+		return ret;
+
	}
```

Comment:
Be carefull with the git config, to avoid CRLF issues please run the following commands on your environment:

```Bash
git config --global core.eol lf
git config --global core.autocrlf input
```